tests/test_models/test_mutables/test_diffchoiceroute.py

```diff
@@ -29,8 +30,10 @@ def test_forward_arch_param(self):
 
         # test with_arch_param = True
         diffchoiceroute = MODELS.build(diff_choice_route_cfg)
+        mutator = DiffModuleMutator()
+        mutator.prepare_from_supernet(diffchoiceroute)
 
-        arch_param = diffchoiceroute.build_arch_param()
+        arch_param = mutator.build_arch_param(num_choices=len(edges_dict))
```

Comment:
Use user-defined instead  of `mutator`  to construct `arch_param`, as we hope the mutable could work(build and forward) independently in our design.